packages/m2m-gateway/src/model/errors.ts

```diff
@@ -38,3 +40,24 @@ export function unexpectedDelegationKind(
     title: "Unexpected Delegation Kind",
   });
 }
+
+export function purposeNotFound(
+  purposeId: purposeApi.Purpose["id"]
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Purpose ${purposeId} not found`,
+    code: "purposeNotFound",
+    title: "Purpose not found",
+  });
+}
+
+export function missingActivePurposeVersion(
+  purposeId: purposeApi.Purpose["id"],
+  logger: Logger
+): ApiError<ErrorCodes> {
+  const error = purposeNotFound(purposeId);
+  logger.warn(
+    `Root cause for Error "${error.title}": there is no active version for purpose ${purposeId}`
+  );
+  return error;
```

Comment:
Let's lift this logic in the converter, to keep error functions clean (see other comment https://github.com/pagopa/interop-be-monorepo/pull/1814#discussion_r2077530284)